lib/crypto/secp256k1/secp256k1.go

```diff
@@ -36,6 +37,16 @@ type PublicKey struct {
 	key ecdsa.PublicKey
 }
 
+// SignVerify verify signature given pubkey and msg
+func SignVerify(pubkey, sig, msg []byte) (bool, error) {
+	ok := secp256k1.VerifySignature(pubkey, msg, sig)
+	if !ok {
+		return false, fmt.Errorf("failed to verify secp256k1 signature")
```

Comment:
Please define this error as a global scope variable at the top of the file

```go
var ErrSignatureVerificationFailed = errors.New("failed to verify signature")
```

and you can use it here with

```go
return false, fmt.Errorf("secp256k1: %w: for message 0x%x, signature 0x%x and public key 0x%x",
     ErrSignatureVerificationFailed, msg, sig, pubkey)
```

This applies to other uses of errors in other signature verification functions